lua/nvim-treesitter/install.lua

```diff
@@ -387,6 +387,7 @@ local function run_install(cache_folder, install_folder, lang, repo, with_sync,
       cmd = function()
         for _, buf in ipairs(vim.api.nvim_list_bufs()) do
           if parsers.get_buf_lang(buf) == lang then
+            vim.treesitter.language.require_language(lang)
```

Comment:
when we do a `vim._ts_remove_language(lang)` even hot-switching parsers works. Like when you had an incompatible parser installed and then install the new one. It would work in the same session. Neovim will leak the module handles anyways :smile: 

Normally, you would need to keep track of lib here until the last parser gets garbage collected until you can `dlclose` the lib.
https://github.com/neovim/neovim/blob/6f25623e793e058c78481b7fb7a42eeff642694b/src/nvim/lua/treesitter.c#L221-L163

Reply:
So, when you see those weird red errors on a incompatible parser, you can to TSInstall or autoinstall and it will get fixed in the same session. We're reattaching to all buffers, so the new parser gets used. If anyone keeps around a reference to an old parser, it is still legal to use it as the lib-handle just got leaked.